tests/cupyx_tests/scipy_tests/stats_tests/test_distributions.py

```diff
@@ -124,11 +124,12 @@ def _entropy(self, xp, scp, dtype, shape, use_qk, base, axis, normalize):
         return res
 
     @testing.for_all_dtypes(no_complex=True)
-    @testing.numpy_cupy_allclose(rtol={cupy.float16: 1e-3,
-                                       cupy.float32: 1e-6,
+    @testing.numpy_cupy_allclose(rtol={cupy.float32: 1e-6,
                                        'default': 1e-15},
                                  scipy_name='scp')
     def test_entropy(self, xp, scp, dtype):
+        # `entropy` returns float32 value for float16 input so float32
+        # tolerance is used for it.
```

Comment:
Are the intermediate computations done in float32?
If that is the case this is fine. But if all the steps are done in float16, even though the output is float32, there will be precision losses. So the original tolerance should be kept.

Reply:
In SciPy, float16 inputs do not seem to be explicitly cast to float32, so the normalization step that divides by a sum over the array will use float16. Thus, the normalization step here should be more accurate than in SciPy. The later use of the `scipy.special` functions does result in things becoming float32, though (I don't think any functions in `scipy.special` support half precision)